src/Middleware/RequestAuthorizationMiddleware.php

```diff
@@ -96,7 +98,12 @@ public function process(ServerRequestInterface $request, RequestHandlerInterface
 
         $result = $service->canResult($identity, $this->getConfig('method'), $request);
         if (!$result->getStatus()) {
-            throw new ForbiddenException($result);
+            $redirectUrl = $this->getConfig('unauthorizedRedirect');
+            if (!empty($redirectUrl)) {
```

Comment:
Cloaking not necessary here, 

    if ($redirectUrl) {

suffices and is identical